wdl2cwl/tests/test_cwl.py

```diff
@@ -18,11 +19,44 @@ def get_file(path: str) -> str:
         ),
     ],
 )
-def test_wdls(wdl_path: str, cwl_path: str) -> None:
-    """Test WDL to CWL conversion."""
-    convertedStr = wdl.main([get_file(wdl_path)])
-    testStr = ""
-    with open(get_file(cwl_path)) as file:
-        testStr = file.read()
-
-    assert convertedStr == testStr
+class TestParameterized:
+    """Contains the test functions for WDL to CWL conversion."""
+
+    def test_wdls(self, wdl_path: str, cwl_path: str) -> None:
+        """Test WDL to CWL conversion."""
+        convertedStr = wdl.convert(get_file(wdl_path))
+        testStr = ""
+        with open(get_file(cwl_path)) as file:
+            testStr = file.read()
+
+        assert convertedStr == testStr
+
+    def test_wdls_patch(
+        self,
+        tmp_path: pathlib.Path,
+        monkeypatch: pytest.MonkeyPatch,
+        wdl_path: str,
+        cwl_path: str,
+    ) -> None:
+        """Test WDL to CWL conversion with patch."""
+        d = tmp_path / "sub"
+        d.mkdir()
+        p = d / "result.cwl"
+
+        monkeypatch.setattr(
+            "sys.argv",
+            [
+                "python",
+                get_file(wdl_path),
+                "--output",
+                get_file(str(p)),
+            ],
+        )
+
+        wdl.main()
+
+        testStr = ""
+        with open(get_file(cwl_path)) as file:
+            testStr = file.read()
+
+        assert p.read_text() == testStr
```

Comment:
Since you now have two files on disk, you can use `filecmp.cmp`: https://docs.python.org/3.6/library/filecmp.html#filecmp.cmp